DapperLib/DapperAOT
# chore: implement integration testing

So far the only testing available is "eye-testing":
1) write user code input
2) generate intercepted code out of it
3) check if generated code makes sense.

It would be great to have some integration tests **using generated bits** against some database to ensure code works. I created `Dapper.AOT.Test.Integration` project as a launcher of tests and `Dapper.AOT.Test.Integration.Executables` project as user code (similar to input we had before).

Tests work in this way:
- Use a type from `Executables` (I will further call it "user code") and parse the code as text
- Create compilation out of it which includes minimum amount of assembly references to make it work
- Run `DapperInterceptorGenerator` on that compilation and get new compilation instance with both user code and generated bits
- Run `compilation.Emit()` to get a built-in assembly in the code
- Via reflection access the user code type specified on the first step and activate an instance of the class
- Execute the code passing pre-created `IDbConnection`. 

To make sure that generated interception code is actually being invoked, I also adjusted `DapperInterceptorGenerator` to include such a code line: 
```c#
global::Dapper.AOT.Test.Integration.Executables.Recording.InterceptorRecorderResolver.Resolve().Record();
```
Since `Dapper.AOT.Test.Integration.Executables` assembly is anyway referenced to the compilation built during a test execution, we have access to all types there. `InterceptorRecorderResolver` is a container for `IInterceptorRecorder` -> test code can register any implementation of recorder to collect all necessary information it can. The implementation I have provided does the following:
- get the StackTrace and takes only last 15 frames out of it
- finds invocation of intercepted code (generated bits)
- finds invocation of user code
- ensures that user code is invoked before intercepted code (which is logical)

In such a manner we can be sure that generated code is actually executed.

Closes #122
